rajawali/src/main/java/c/org/rajawali3d/engine/EngineImpl.java

```diff
@@ -220,105 +244,104 @@ public void onRenderSurfaceCreated(EGLConfig config, GL10 gl, int width, int hei
 
     @Override
     public void onRenderSurfaceDestroyed(SurfaceTexture surface) {
-        stopRendering();
+        stopFrames();
     }
 
     @Override
     public void onRenderSurfaceSizeChanged(GL10 gl, int width, int height) {
-        defaultViewportWidth = width;
-        defaultViewportHeight = height;
+        surfaceWidth = width;
+        surfaceHeight = height;
 
-        synchronized (renderables) {
-            for (Renderable renderable : renderables) {
-                renderable.onRenderSurfaceSizeChanged();
+        synchronized (surfaceCallbacks) {
+            for (SurfaceCallback surfaceListener : surfaceCallbacks) {
+                surfaceListener.onSurfaceSizeChanged(width, height);
             }
         }
 
-        startRendering();
+        startFrames();
     }
 
+    /**
+     * TODO Override this method in a stereo renderer to call renderViews() for each Eye
+     *
+     * @param gl {@link GL10} for rendering.
+     */
     @Override
     public void onRenderFrame(GL10 gl) {
-        synchronized (nextRenderableLock) {
-            // Check if we need to switch the scene, and if so, do it.
-            if (nextRenderable != null) {
-                switchRenderable(nextRenderable);
-                nextRenderable = null;
-            }
-        }
 
         final long currentTime = System.nanoTime();
-        final long elapsedRenderTime = currentTime - renderStartTime;
-        final double deltaTime = (currentTime - lastRender) / 1e9;
-        lastRender = currentTime;
+        final double deltaTime = (currentTime - lastFrameTime) / 1e9;
+        lastFrameTime = currentTime;
 
-        try {
-            onRender(elapsedRenderTime, deltaTime);
-        } catch (InterruptedException e) {
-            Log.e(TAG, "Frame skipped due to thread interruption.");
-        }
+        startFrame(deltaTime);
```

Comment:
I removed it because in general, frames start/stop are independent of the Scene or SceneView lifecycles, and passing this around seemed unnecessary. I did add a getter among the frame control methods, but if there is a use case for passing it around then it makes sense.